test/test_messaging.py

```diff
@@ -411,3 +381,264 @@ def spawn_managed_thread(method):
         consumer.stop()
 
     consumer.queue_consumer.kill()
+
+
+@skip_if_no_toxiproxy
+class TestPublisherDisconnections(object):
+    """ Test and demonstrate behaviour under poor network conditions.
+
+    Publisher confirms must be enabled for some of these tests to pass. Without
+    confirms, previously used but now dead connections will accept writes
+    without raising. These tests are skipped in this scenario.
+
+    Note that publisher confirms do not protect against sockets that remain
+    open but do not deliver messages (i.e. `toxiproxy.timeout(0)`).
+    This can only be mitigated with AMQP heartbeats (not yet supported)
+    """
+
+    @pytest.fixture
+    def tracker(self):
+        return Mock()
+
+    @pytest.yield_fixture(autouse=True)
+    def toxic_publisher(self, toxiproxy):
+        with patch.object(Publisher, 'amqp_uri', new=toxiproxy.uri):
+            yield
+
+    @pytest.yield_fixture(autouse=True, params=[True, False])
+    def use_confirms(self, request):
+        with patch.object(Publisher, 'use_confirms', new=request.param):
+            yield request.param
+
+    @pytest.yield_fixture
+    def publisher_container(
+        self, request, container_factory, tracker, rabbit_config
+    ):
+        retry = False
+        if "enable_retry" in request.keywords:
```

Comment:
Strict mode actually works fine if you register `enable_retry` as a mark in an ini file.

Is there a better way to control the behaviour of a fixture like this? Also do you want me to enable strict mode as part of this PR?